src/infra/main/template.yaml

```diff
@@ -177,6 +182,19 @@ Resources:
             Condition:
               ArnLike:
                 aws:SourceArn: !Sub "arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:*"
+          - Effect: Allow
+            Principal:
+              Service: sns.amazonaws.com
+            Action:
+              - "kms:Decrypt"
+              - "kms:GenerateDataKey"
+            Resource: "*"
+            Condition:
+              ArnEquals:
+                aws:SourceArn: !If
+                  - IsInternal
+                  - !Ref AccountDeletionProcessorSNSTopic
+                  - !FindInMap [ EnvConfig, !Ref Environment, AccountDeletionProcessorSNSTopicARN ]
```

Comment:
I think this will make the key inaccessible by sns given that we are passing empty strings for any other env but dev.. is this a good idea? @domantas-gds

Reply:
Yes as discussed let's Ref the AccountDeletionProcessorSNSTopic we've defined until we can populate those mappings :) So will also need to remove the IsInternal condition from the AccountDeletionProcessorSNSTopic resource